src/status_im2/contexts/syncing/scan_sync_code/view.cljs

```diff
@@ -255,136 +252,102 @@
   [insets translate-y]
   [:f> f-bottom-view insets translate-y])
 
-(defn- check-qr-code-data
-  [event]
+(defn- check-qr-code-and-navigate
+  [event set-qr-code-succeeded]
   (let [connection-string        (string/trim (oops/oget event "nativeEvent.codeStringValue"))
         valid-connection-string? (sync-utils/valid-connection-string? connection-string)]
     (if valid-connection-string?
-      (debounce/debounce-and-dispatch [:syncing/input-connection-string-for-bootstrapping
-                                       connection-string]
-                                      300)
+      (do
+        (set-qr-code-succeeded)
+        (debounce/debounce-and-dispatch [:syncing/input-connection-string-for-bootstrapping
+                                         connection-string]
+                                        300))
       (rf/dispatch [:toasts/upsert
                     {:icon           :i/info
                      :icon-color     colors/danger-50
                      :override-theme :light
                      :text           (i18n/label :t/error-this-is-not-a-sync-qr-code)}]))))
 
-(defn render-camera
-  [show-camera? torch-mode qr-view-finder camera-ref on-read-code]
-  (when (and show-camera? (:x qr-view-finder))
-    [:<>
-     [rn/view {:style style/camera-container}
-      [camera-kit/camera
-       {:ref          #(reset! camera-ref %)
-        :style        style/camera-style
-        :camera-type  camera-kit/camera-type-back
-        :zoom-mode    :off
-        :torch-mode   torch-mode
-        :scan-barcode true
-        :on-read-code on-read-code}]]
-     [hole-view/hole-view
-      {:style style/hole
-       :holes [(merge qr-view-finder
-                      {:borderRadius 16})]}
-      [blur/view
-       {:style            style/absolute-fill
-        :blur-amount      10
-        :blur-type        :transparent
-        :overlay-color    colors/neutral-80-opa-80
-        :background-color colors/neutral-80-opa-80}]]]))
+(defn- render-camera
+  [{:keys [torch-mode qr-view-finder set-qr-code-succeeded]}]
+  [:<>
+   [rn/view {:style style/camera-container}
+    [camera-kit/camera
+     {:style        style/camera-style
+      :camera-type  camera-kit/camera-type-back
+      :zoom-mode    :off
+      :torch-mode   torch-mode
+      :scan-barcode true
+      :on-read-code #(check-qr-code-and-navigate % set-qr-code-succeeded)}]]
+   [hole-view/hole-view
+    {:style style/hole
+     :holes [(merge qr-view-finder {:borderRadius 16})]}
+    [blur/view
+     {:style            style/absolute-fill
+      :blur-amount      10
+      :blur-type        :transparent
+      :overlay-color    colors/neutral-80-opa-80
+      :background-color colors/neutral-80-opa-80}]]])
```

Comment:
Small refactor:

This `render-camera` component has a `when` and it could return `nil`. Instead of creating a component that is returning nil, now `render-camera` always renders the camera, and the condition to show it or not is when calling it.